### PR TITLE
feat: configuration form changes

### PR DESCRIPTION
**Pull request by TuiMao233**
related: #56

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ For a example, you can see [demo/yun](./demo/yun/) folder.
 
 - ⚡️ [Vue 3](https://github.com/vuejs/vue-next), [Vite 2](https://github.com/vitejs/vite), [pnpm](https://pnpm.js.org/), [ESBuild](https://github.com/evanw/esbuild) - born with fastness
 - 🔥 Hot Reload with Config & Markdown
-- 🔧 Type Tooltip for all config by `valaxy.config.ts`
+- 🔧 Type Tooltip for all config by `valaxy.config.ts` or `site.config.ts`
```
Comment (YunYouJun) [inline README.md]: Maybe we can use `and`.

File: docs/blog.config.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { defineConfig } from 'valaxy'
+import { defineSite } from 'valaxy'
```
Comment (YunYouJun) [inline docs/blog.config.ts]: File name should be `site.config.ts`.

File: docs/pages/theme/comment.md [hunk 3]
```diff
@@ -18,10 +18,10 @@ Valaxy 决定中心化地提供各类封装好的评论钩子函数。
 
 ```vue {2}
 <script lang="ts" setup>
-import { useConfig, useWaline } from 'valaxy'
+import { useSiteConfig, useWaline } from 'valaxy'
 
 // 读取用户配置
-const config = useConfig()
+const config = useSiteConfig()
```
Comment (YunYouJun) [inline docs/pages/theme/comment.md]: We use `defineSite`, so i think we should use `useSite` for symmetry.

File: packages/create-valaxy-theme/index.js [hunk 4]
```diff
@@ -36,8 +36,8 @@ async function init() {
   console.log(`  ${bold('Check it')}:`)
   console.log()
   console.log(`- Change ${bold('author')} name in ${yellow('LICENSE')} & ${green('package.json')} & ${blue('.github')}`)
-  console.log(`- Change ${blue('valaxy.config.ts')} theme: ${yellow('starter')} to ${cyan(`${themeName}`)}`)
-  console.log(`- Rename ${yellow('valaxy-theme-starter')} to ${cyan(`valaxy-theme-${themeName}`)}`)
+  console.log(`- Change ${blue('site.config.ts')} theme: ${yellow('starter')} to ${cyan(`${themeName}`)}`)
+  console.log(`- Rename ${yellow(`valaxy-theme-${themeName}`)} to ${cyan(`valaxy-theme-${themeName}`)}`)
```
Comment (YunYouJun) [inline packages/create-valaxy-theme/index.js:40]: The former is `starter`.

File: packages/valaxy-theme-press/components/PressNav.vue [hunk 5]
```diff
@@ -1,13 +1,13 @@
 <script lang="ts" setup>
-import { isDark, toggleDark, useConfig } from 'valaxy'
+import { isDark, toggleDark, useSiteConfig } from 'valaxy'
```
Comment (YunYouJun) [inline packages/valaxy-theme-press/components/PressNav.vue]: Similarly `useSite`.

File: packages/valaxy/client/config.ts [hunk 6]
```diff
@@ -24,16 +23,16 @@ interface ValaxyContext {
   userRoot: string
 }
 
-export const valaxyConfigSymbol: InjectionKey<ComputedRef<ValaxyConfig>> = Symbol('valaxy:config')
-export const valaxyConfigRef = shallowRef<ValaxyConfig>(parse<ValaxyConfig>(valaxyConfig))
+export const ValaxySiteConfigSymbol: InjectionKey<ComputedRef<ValaxySiteConfig>> = Symbol('valaxy:config')
```
Comment (YunYouJun) [inline packages/valaxy/client/config.ts]: `valaxySiteSymbol` and `valaxy:site`

Variables start with lowercase letter.
Type start with uppercase letter.

File: packages/valaxy/node/options.ts [hunk 7]
```diff
@@ -48,10 +47,13 @@ export interface ValaxyPluginOptions {
...
   }) => void
+  plugins?: ValaxyPluginOption[]
 }
-
-export interface ValaxyThemePlugin extends ValaxyPluginOptions {
-  vite?: Omit<UserConfig, 'valaxy'>
+export type ValaxyPluginLike = ValaxyPlugin | ValaxyPlugin[] | false | null | undefined
+export type ValaxyPluginOption = ValaxyPluginLike | string | [string, any]
+export interface ValaxyPlugin extends ValaxyConfig {
+  enhanceAppFiles?: string[]
```
Comment (YunYouJun) [inline packages/valaxy/node/options.ts]: We do not need `enhanceAppFIles` & `globalUIComponents`.
I think we should just use `setup/main.ts` & `components` folder.

Comment (hairyf) [reply packages/valaxy/node/options.ts]: As for `setup/main.ts` | `components`, I'm not very comfortable with it, it works great for theme development and user usage

But there is a certain degree of counter-intuitiveness in the performance of addon

**Review by YunYouJun — APPROVED** [review]

**Pull request merged**